### PR TITLE
Feat/en 2678 data and execution module

### PR DESCRIPTION
**Pull request by sasurobert**
renamed smart contract result to unsigned transactions.

### CODE REVIEW

File: process/unsigned/interceptedUnsignedTransaction.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
+package unsigned
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data"
```
Comment (iulianpascalau) [inline process/unsigned/interceptedUnsignedTransaction.go]: imports are not sorted

File: dataRetriever/errors.go [hunk 2]
```diff
@@ -61,8 +61,8 @@ var ErrNilMessenger = errors.New("nil Messenger")
 // ErrNilTxDataPool signals that a nil transaction pool has been provided
 var ErrNilTxDataPool = errors.New("nil transaction data pool")
 
-// ErrNilSmartContractResultsPool signals that a nil smart contract results pool has been provided
-var ErrNilSmartContractResultsPool = errors.New("nil smart contract results data pool")
+// ErrNilUnsignedTransactionPool signals that a nil unsigned transactions pool has been provided
+var ErrNilUnsignedTransactionPool = errors.New("nil smart contract results data pool")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply dataRetriever/errors.go]: done

File: process/errors.go [hunk 3]
```diff
@@ -364,14 +364,14 @@ var ErrNilSmartContractResult = errors.New("smart contract result is nil")
 // ErrInvalidDataInput signals that the data input is invalid for parsing
 var ErrInvalidDataInput = errors.New("data input is invalid to create key, value storage output")
 
-// ErrNoSmartContractResultInMessage signals that message does not contain required data
-var ErrNoSmartContractResultInMessage = errors.New("no smart contract result in message")
+// ErrNoUnsignedTransactionInMessage signals that message does not contain required data
+var ErrNoUnsignedTransactionInMessage = errors.New("no smart contract result in message")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/errors.go]: done

File: process/errors.go [hunk 4]
```diff
@@ -364,14 +364,14 @@ var ErrNilSmartContractResult = errors.New("smart contract result is nil")
...
 
-// ErrNilScrDataPool signals that smart contract pool is nil
-var ErrNilScrDataPool = errors.New("smart contract result pool is nil")
+// ErrNilUTxDataPool signals that unsigned transaction pool is nil
+var ErrNilUTxDataPool = errors.New("smart contract result pool is nil")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/errors.go]: done

File: process/errors.go [hunk 5]
```diff
@@ -364,14 +364,14 @@ var ErrNilSmartContractResult = errors.New("smart contract result is nil")
...
 
-// ErrNilScrStorage signals that smart contract storage is nil
-var ErrNilScrStorage = errors.New("smart contract result storage is nil")
+// ErrNilUTxStorage signals that unsigned transaction storage is nil
+var ErrNilUTxStorage = errors.New("smart contract result storage is nil")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/errors.go]: done

File: process/unsigned/interceptedUnsignedTransaction.go (hunk 1 above)
Comment (sasurobert) [reply process/unsigned/interceptedUnsignedTransaction.go]: done

File: cmd/node/factory/structs.go [hunk 6]
```diff
@@ -838,7 +851,7 @@ func createShardDataPoolFromConfig(
 		return nil, err
 	}
 
-	scrPool, err := shardedData.NewShardedData(getCacherFromConfig(config.SmartContractDataPool))
+	uTxPool, err := shardedData.NewShardedData(getCacherFromConfig(config.SmartContractDataPool))
```
Comment (bogdan-rosianu) [inline cmd/node/factory/structs.go]: Rename this configuration

Comment (sasurobert) [reply cmd/node/factory/structs.go]: done

**Pull request merged**